plugins/pixart-rf/fu-pxi-firmware.c

```diff
@@ -51,13 +56,30 @@ fu_pxi_firmware_check_magic(FuFirmware *firmware, GBytes *fw, gsize offset, GErr
 		return FALSE;
 	}
 	if (magic != PIXART_RF_FW_HEADER_MAGIC) {
-		g_set_error(error,
-			    FWUPD_ERROR,
-			    FWUPD_ERROR_INVALID_FILE,
-			    "invalid magic, expected 0x%08X got 0x%08X",
-			    (guint32)PIXART_RF_FW_HEADER_MAGIC,
-			    (guint32)magic);
-		return FALSE;
+		/* if the magic number is not found, then start from the 821st byte from the end for
+		 * HPAC header */
+		if (!fu_memread_uint64_safe(g_bytes_get_data(fw, NULL),
+					    g_bytes_get_size(fw),
+					    g_bytes_get_size(fw) -
+						PIXART_RF_FW_HEADER_HPAC_POS_FROM_END +
+						PIXART_RF_FW_HEADER_TAG_OFFSET,
+					    &magic,
+					    G_BIG_ENDIAN,
+					    error)) {
+			g_prefix_error(error, "failed to read magic: ");
+			return FALSE;
+		}
+		if (magic != PIXART_RF_FW_HEADER_MAGIC) {
+			g_set_error(error,
+				    FWUPD_ERROR,
+				    FWUPD_ERROR_INVALID_FILE,
+				    "invalid magic, expected 0x%08X got 0x%08X",
+				    (guint32)PIXART_RF_FW_HEADER_MAGIC,
+				    (guint32)magic);
+			return FALSE;
+		} else {
```

Comment:
`else` is redundant; when `if` block passes function returns `FALSE`.

Reply:
solved in 269968b90c62b0300ff7b0ef900798a43124ff41